.github/workflows/main.yml

```diff
@@ -0,0 +1,13 @@
+name: Static code analysis
+
+on: [push]
```

Comment:
Should be on `push` and `pull_request`.

Reply:
I still didn't find a situation where solely `push` didn't run the tests. Adding them both on the contrary will run the tests twice when you open a PR.

Reply:
If you do a PR from a fork, then this will action will never run. 

Feel free to merge this now and do a PR from a fork later to try yourself.